mixer/pkg/il/evaluator/evaluator.go

```diff
@@ -206,20 +201,16 @@ func (ctx *attrContext) getOrCreateCacheEntry(expr string, functions map[string]
 // DefaultCacheSize is the default size for the expression cache.
 const DefaultCacheSize = 1024
 
-// DefaultMaxStringTableSizeForPurge is the default value for
-const DefaultMaxStringTableSizeForPurge = 1024
-
 // NewILEvaluator returns a new instance of IL.
-func NewILEvaluator(cacheSize int, maxStringTableSizeForPurge int) (*IL, error) {
+func NewILEvaluator(cacheSize int) (*IL, error) {
 	// check the cacheSize here, to ensure that we can ignore errors in lru.New calls.
 	// cacheSize restriction is the only reason lru.New returns an error.
 	if cacheSize <= 0 {
 		return nil, errors.New("cacheSize must be positive")
 	}
 
 	return &IL{
-		cacheSize:                  cacheSize,
-		maxStringTableSizeForPurge: maxStringTableSizeForPurge,
-		functions:                  expr.FuncMap(runtime.ExternFunctionMetadata),
+		cacheSize: cacheSize,
```

Comment:
I really don't like "cacheSize" as a name since it's ambiguous between "size in bytes" and "size in entries". maxCacheEntries is usually a better choice for this kind of name IMO.

Reply:
Noted. Hopefully, this will not survive very long, if I can land the CompiledExpression work soon.